force-app/main/default/triggers/OpportunityTrigger.trigger

```diff
@@ -0,0 +1,38 @@
+trigger OpportunityTrigger on Opportunity (before update, before Delete) {
+    // Validate that the amount is greater than 5000.
+    //  Prevent the deletion of a closed won opportunity for a banking account.
+    // Set the primary contact on the opportunity to the contact with the title of CEO.
+    
+    if (Trigger.isBefore && Trigger.isUpdate) {
+        
+        // Get a list of Account Ids from the Opportunities
+        Set<Id> accountIds = new Set<Id>();
+        for (Opportunity opp : Trigger.new) {
+            accountIds.add(opp.AccountId);
+        }
+
+        // Query for Contacts with the Title of 'CEO' on these Accounts
+        Map<Id, Contact> ceoContacts = new Map<Id, Contact>();
+        for(Contact c : [SELECT Id, AccountId FROM Contact WHERE AccountId IN :accountIds AND Title = 'CEO']) {
+            ceoContacts.put(c.AccountId, c);
+        }
```

Comment:
You should move these logic into the trigger handler so there's limited logic in the trigger